Automattic/wp-calypso
# Modularise login state

This PR is one of many working on modularising state in Calypso. This one handles login.

For mode details on state modularisation, see #39261 and p4TIVU-9lM-p2.

After this PR, the `login` reducer should no longer be loaded as part of the critical path on most routes, as it moves away from the `entry-main` chunk. It's a small saving (~ 1.5KB compressed), but the goal of state modularisation is to produce cumulative effects over many PRs.

#### Changes proposed in this Pull Request

* Modularise login state
* Split login actions into individual modules
* Rework dependencies between login actions and login-related `http-data` modules

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/read` with a full page refresh.
* Click `State` on the top right of the Redux tab in DevTools to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and note that there's no `login` key, even if you expand the list of keys. This indicates that the login state hasn't been loaded yet.
* Click the user avatar on the masterbar, followed by `Security` on the sidebar.
* Verify that a `login` key is added with the login state. This indicates that the login state has now been loaded.
* Verify that login works normally, including 2FA functionality. This indicates that I didn't mess up.
